### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ivan-shpynda**

### CODE REVIEW

**Review by leonid-vegera — APPROVED** [review]
Good job!

File: src/transformState.js [hunk 1]
```diff
@@ -61,8 +61,25 @@
  * @param {Object} state
  * @param {Object[]} transforms
  */
+
 function transformState(state, transforms) {
-  // write code here
+  for (const obj of transforms) {
```
Comment (VinnieJ-2k20) [inline src/transformState.js:66]:
```suggestion
  for (const transform of transforms) {
```

File: src/transformState.js [hunk 2]
```diff
@@ -61,8 +61,25 @@
...
+    if (obj['operation'] === 'addProperties') {
```
Comment (VinnieJ-2k20) [inline src/transformState.js:67]:
```suggestion
    if (obj.operation === 'addProperties') {
```

File: src/transformState.js [hunk 3]
```diff
@@ -61,8 +61,25 @@
...
+      Object.assign(state, obj['properties']);
```
Comment (VinnieJ-2k20) [inline src/transformState.js:68]:
```suggestion
      Object.assign(state, obj.properties);
```

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Let's rewrite the solution using `switch/case` structure.

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
`switch/case` structure should always have default option in case of irregular inputs.
In this task it may do nothing, or throw an error.

**Review by VinnieJ-2k20 — APPROVED** [review]

**Pull request closed without merging**